update_command.php

```diff
@@ -62,39 +67,99 @@
+        if ($input->getOption('update-prs')) {
+            // checkout latest existing pr branch
+            cmd('git fetch pr-remote', $MODULE_DIR);
+            $allBranches = explode("\n", cmd('git branch -r', $MODULE_DIR));
+            // example branch name: pulls/5/module-standardiser-1691550112
+            $allBranches = array_map('trim', $allBranches);
+            $allBranches = array_filter($allBranches, function($branch) {
+                return preg_match('#^pr\-remote/pulls/[0-9\.]+/module\-standardiser\-[0-9]{10}$#', $branch);
+            });
+            if (empty($allBranches)) {
+                warning("Could not find an existing PR branch for $repo - skipping");
+                continue;
+            }
+            // sort so that the branch with the highest timestamp goes to position 0 in the array
+            usort($allBranches, function($a, $b) {
+                return (substr($a, -10) <=> substr($b, -10)) * -1;
+            });
+            $branchToCheckout = $allBranches[0];
+            $branchToCheckout = preg_replace('#^pr\-remote/#', '', $branchToCheckout);
+            $prBranch = $branchToCheckout;
+            $allPRs = github_api("https://api.github.com/repos/$account/$repo/pulls?per_page=100");
+            $allPRs = array_filter($allPRs, function($pr) use($prBranch) {
+                 return $pr['title'] === PR_TITLE && $pr['head']['ref'] === $prBranch && $pr['state'] === 'open';
+            });
+            if (count($allPRs) < 1) {
+                warning("Could not find an existing open PR for $repo for branch $prBranch - skipping");
+                continue;
+            }
+        } else {
+            // get all branches
+            $allBranches = explode("\n", cmd('git branch -r', $MODULE_DIR));
+            $allBranches = array_map(fn($branch) => trim(str_replace('origin/', '', $branch)), $allBranches);
+
+            // reset to the default branch so that we can then calculate the correct branch to checkout
+            // this is needed for scenarios where we may be on something unparsable like pulls/5/lorem-ipsum
+            $cmd = "git symbolic-ref refs/remotes/origin/HEAD | sed 's@^refs/remotes/origin/@@'";
+            $defaultBranch = cmd($cmd, $MODULE_DIR);
+            cmd("git checkout $defaultBranch", $MODULE_DIR);
+
+            // checkout the branch to run scripts over
+            $currentBranch = cmd('git rev-parse --abbrev-ref HEAD', $MODULE_DIR);
+            // ensure that we're on a standard next-minor style branch
+            if (!ctype_digit($currentBranch)) {
+                $tmp = array_filter($allBranches, fn($branch) => ctype_digit($branch));
+                if (empty($tmp)) {
+                    error('Could not find a next-minor style branch');
+                }
+                $currentBranch = max($tmp);
+                cmd("git checkout $currentBranch", $MODULE_DIR);
+            }
```

Comment:
This seems sensible - except that this means any changes which _need_ to be on the default branch to work (e.g. changes to cron-based workflows) simply won't do anything. Do you have any thoughts about those scenarios? Fluent for example has `master` as its default branch.

Reply:
Our standard is integer's for next-minors. If fluent doesn't play nice that's because fluent is non-standard and the solution is to standardise fluent, not to loosen things here

Reply:
I'm on the fence about this, since I don't think it's likely we'll ever get the maintainer to follow our branching strategy. But that's a much wider discussion than should be held on this PR so I'm okay with leaving this as is for now.